specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/stable/2021-08-09/baremetalinfrastructure.json

```diff
@@ -0,0 +1,654 @@
+                    },
+                    {
+                        "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/AzureBareMetalInstancesListResult"
+                        }
+                    },
+                    "default": {
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    }
+                },
+                "x-ms-pageable": {
+                    "nextLinkName": "nextLink"
+                }
+            }
+        },
+        "/providers/Microsoft.BareMetalInfrastructure/operations": {
+            "get": {
+                "tags": [
+                    "AzureBareMetal"
+                ],
+                "operationId": "Operations_List",
+                "x-ms-examples": {
+                    "List all management operations supported by the AzureBareMetal RP": {
+                        "$ref": "./examples/AzureBareMetalOperations_List.json"
+                    }
+                },
+                "description": "Gets a list of AzureBareMetal management operations.",
+                "consumes": [
+                    "application/json"
+                ],
+                "produces": [
+                    "application/json"
+                ],
+                "parameters": [
+                    {
+                        "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/OperationList"
+                        }
+                    },
+                    "default": {
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    }
+                },
+                "x-ms-pageable": {
+                    "nextLinkName": null
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.BareMetalInfrastructure/bareMetalInstances/{azureBareMetalInstanceName}": {
```

Comment:
If this is a tracked resource how is it created without a PUT request and how are they deleted without the DELETE request that was in 2021-08-06-preview?

Reply:
I'm sure this has probably come up before.  May want to boilerplate call this info out in any future reviews

Reply:
Our service's flow is a little unique, as our resources' creation is negotiated manually via contract rather than just calling a PUT API for any given resource. We are looking at enabling the regular ARM PUT flow instead, but would add it in a future GA version as it is not reliable yet.

Reply:
Similarly, DELETE is not supported because the contract's terms are for set durations of time, and thus the customer cannot choose (and call an API) when to delete resources outside of that.

Reply:
These are marked as tracked resources in your RP manifest.  How does the [ARM cache](https://armwiki.azurewebsites.net/introduction/concepts/caching.html?q=arm%20cache) (and Azure Resource Graph, etc...) stay up to date with their state?  If you don't support a DELETE what happens when a user tries to remove a resource group that contains one of these (the RG deletion would never succeed since ARM can't delete the resource).

Reply:
We have internal logic to rehydrate the ARM cache when resources are deleted in the backend.

Correct, since we don't support delete, we also do not support deletion of a resource group that contains this resource type.

We have discussed an exception for supporting PUT/DELETE, noted in our tracking work item here: https://dev.azure.com/msazure/One/_workitems/edit/9821575

Reply:
Please include this information up front in future swagger PRs.  It'll make things a lot faster